internal/controller/api/job_receiver.go

```diff
@@ -57,22 +114,9 @@ func (jr *JobReceiver) handleJob() http.HandlerFunc {
 
 		var jobRequest JobRequest
 
-		body, err := ioutil.ReadAll(io.LimitReader(req.Body, 1048576))
-
-		if err != nil {
-			panic(err)
-		}
-
-		if err := req.Body.Close(); err != nil {
-			panic(err)
-		}
-
-		if err := json.Unmarshal(body, &jobRequest); err != nil {
-			w.Header().Set("Content-Type", "application/json; charset=UTF-8")
-			w.WriteHeader(http.StatusUnprocessableEntity)
-			if err := json.NewEncoder(w).Encode(err); err != nil {
-				panic(err)
-			}
+		if err := decodeJSON(w, req, &jobRequest); err != nil {
+			log.Println(err)
```

Comment:
Continuing the thought from my comment on the decodeJSON method...

We could set the http status and response in one place here.

Reply:
:+1: